tsig.go

```diff
@@ -131,15 +131,11 @@ func TsigGenerate(m *Msg, secret, requestMAC string, timersOnly bool) ([]byte, s
 		return nil, "", ErrKeyAlg
 	}
 	h.Write(buf)
+	// Copy all TSIG fields expect MAC and its size, which are filled using the computed digest.
+	*t = *rr
```

Comment:
we could alternatively copy the error and other data (plus its length) fields separately below.  I chose this way as it seemed more concise and less error prone (it's clear no fields are forgotten and all fields are an exact copy of the original TSIG RR), but I don't mind taking the alternative approach.